src/meta/meta_ckpt.c

```diff
@@ -1273,28 +1465,43 @@ __wt_meta_ckptlist_set(
     WT_CKPT *ckpt;
     WT_DECL_ITEM(buf);
     WT_DECL_RET;
-    const char *fname;
+    WT_ITEM checkpoint_blkmods_buffer;
+    const char *filename;
     bool has_lsn;
 
-    fname = dhandle->name;
-    WT_RET(__wt_scr_alloc(session, 1024, &buf));
+    filename = dhandle->name;
+    WT_CLEAR(checkpoint_blkmods_buffer);
+
+    WT_ERR(__wt_scr_alloc(session, 1024, &buf));
     WT_ERR(__wt_meta_ckptlist_to_meta(session, ckptbase, buf));
     /* Add backup block modifications for any added checkpoint. */
     WT_CKPT_FOREACH (ckptbase, ckpt)
-        if (F_ISSET(ckpt, WT_CKPT_ADD))
+        if (F_ISSET(ckpt, WT_CKPT_ADD)) {
+            WT_CLEAR(checkpoint_blkmods_buffer);
+
             WT_ERR(__wt_ckpt_blkmod_to_meta(session, buf, ckpt));
 
+            if (F_ISSET(dhandle, WT_DHANDLE_IS_METADATA)) {
+                /* Skip as there are no backup blocks allocated for the metadata file */
+            } else {
+                WT_ERR(__check_backup_blocks(session, ckpt, &checkpoint_blkmods_buffer, filename));
```

Comment:
Should all of this be under DIAGNOSTIC? I'm not sure we want to be performing this verification on production builds on every checkpoint? On the other hand, this is not a performant path. I don't know the answer but need to ask the question!

Reply:
I think it's important to perform these checks all the time (unless there's a perf impact) so that we are verifying on production environments, just in case a problem slips in down the track.